plugins/battery/battery.plugin.zsh

```diff
@@ -64,6 +67,53 @@ if [[ "$OSTYPE" = darwin* ]] ; then
     [[ $(ioreg -rc "AppleSmartBattery"| grep '^.*"IsCharging"\ =\ ' | sed -e 's/^.*"IsCharging"\ =\ //') == "Yes" ]]
   }
 
+elif [[ "$OSTYPE" = freebsd*  ]] ; then
+
+		function battery_is_charging() {
+				[[ $(sysctl -n hw.acpi.battery.state) -eq 2 ]]
+		}
+
+function battery_pct() {
+		if (( $+commands[sysctl] )) ; then
+				echo "$(sysctl -n hw.acpi.battery.life)"
+		fi
+}
+
+function battery_pct_remaining() {
+		if [ ! $(battery_is_charging) ] ; then
+				battery_pct
+		else
+				echo "External Power"
+		fi
+}
+
+function battery_time_remaining() {
+		remaining_time=$(sysctl -n hw.acpi.battery.time)
+		if [[ $remaining_time -ge 0 ]] ; then
+				# calculation from https://www.unix.com/shell-programming-and-scripting/23695-convert-minutes-hours-minutes-seconds.html
+				((hour=$remaining_time/60))
+				((minute=$remaining_time-$hour*60))
+				echo $hour:$minute
+		fi
+}
+
+function battery_pct_prompt() {
```

Comment:
Functions are not correctly indented, as it with the rest of the script.

Reply:
fixed